### PR TITLE
enforce code covereage on every run and with low overhead

### PR DESCRIPTION
**Pull request by grosser**
includes 2 of my other PRs, so needs a rebase before it can be merged

 - fixing coverage to make it possible while still using bundler
 - enforcing coverage on every run (super low overhead compared to simplecov)
 - adding tests to get coverage to 100%

Error looks like this:
screen shot 2018-02-03 at 7 39 22 am

@masci

### CODE REVIEW

**Comment by grosser** [comment]
ready to merge

File: Gemfile [hunk 1]
```diff
@@ -1,11 +1,11 @@
 source 'https://rubygems.org'
-gemspec
 
 gem 'rake', '10.1.0'
 gem 'rack', '~> 1.6'
 gem 'minitest'
 gem "yard", "~> 0.8.7.3"
-gem 'simplecov'
+gem 'jeweler', '~> 1.8'
```
Comment (grosser) [inline Gemfile]: jeweler should go away

**Comment by grosser** [comment]
otherwise looks right! :)

File: Gemfile (hunk 1 above)
Comment (grosser) [reply Gemfile]: pushed a fix

**Comment by masci** [comment]
@grosser thanks, going to release soon and ship allthethings

**Comment by grosser** [comment]
got a few more PRs :)

On Sat, Feb 3, 2018 at 2:52 PM, Massimiliano Pippi <notifications@github.com

**Pull request merged**